### PR TITLE
[eclipse/xtext#1384] Convert Jenkinsfile to declarative pipeline

### PR DESCRIPTION
**Pull request by kthoms**
Also add Slack notification

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by kthoms** [comment]
TODO solve test issues

**Comment by kthoms** [comment]
Test failures are on ci-staging, but JIPP is OK. This change is only relevant for JIPP.
https://ci.eclipse.org/xtext/job/xtext-maven/job/kth_issue1384/

**Comment by cdietrich** [comment]
@kthoms 

```
00:57:29  [INFO] --- maven-install-plugin:2.5.2:install-file (self) @ xtext-maven-plugin ---
00:57:29  [INFO] Installing /jobs/genie.xtext/xtext-maven/branches/kt-issue1384.l1sl50/workspace/org.eclipse.xtext.maven.plugin/target/xtext-maven-plugin-2.19.0-SNAPSHOT-it.jar to /jobs/genie.xtext/xtext-maven/branches/kt-issue1384.l1sl50/workspace/.m2/repository/org/eclipse/xtext/xtext-maven-plugin/IT-SNAPSHOT/xtext-maven-plugin-IT-SNAPSHOT.jar
00:57:29  [INFO] Installing /jobs/genie.xtext/xtext-maven/branches/kt-issue1384.l1sl50/workspace/org.eclipse.xtext.maven.plugin/pom.xml to /jobs/genie.xtext/xtext-maven/branches/kt-issue1384.l1sl50/workspace/.m2/repository/org/eclipse/xtext/xtext-maven-plugin/IT-SNAPSHOT/xtext-maven-plugin-IT-SNAPSHOT.pom
```

i am sure the pwd was there for a reason so that the test can find it.
```
<repository>
			<id>it-repo</id>
			<url>file://${maven.repo.local}</url>
			<releases>
				<enabled>true</enabled>
			</releases>
			<snapshots>
				<enabled>true</enabled>
			</snapshots>
		</repository>
```

i am not sure if the filesystems at build and test file are the same relative wise with your way

**Comment by cdietrich** [comment]
here are fails too:
https://ci.eclipse.org/xtext/job/xtext-maven/job/kt_issue1384/1/consoleFull

File: Jenkinsfile [hunk 1]
```diff
@@ -1,27 +1,76 @@
...
-				dir('build') { deleteDir() }
-				dir('.m2/repository/org/eclipse/xtext') { deleteDir() }
-				dir('.m2/repository/org/eclipse/xtend') { deleteDir() }
-			}
-			
-			stage('Maven Build') {
-				def workspace = pwd()
-				def mvnHome = tool 'M3'
-				env.M2_HOME = "${mvnHome}"
-				sh "${mvnHome}/bin/mvn --batch-mode --update-snapshots -fae -PuseJenkinsSnapshots -DJENKINS_URL=$JENKINS_URL -Dmaven.test.failure.ignore=true -Dmaven.repo.local=${workspace}/.m2/repository -f org.eclipse.xtext.maven.parent/pom.xml -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn clean deploy"
```
Comment (cdietrich) [inline Jenkinsfile]: i believe the ${workspace} is important here to get the tests running

Comment (kthoms) [reply Jenkinsfile]: OK, I thought that the build already runs within the workspace and it should be obsolete then.

**Comment by kthoms** [comment]
I'll delete the old branch "kth_issue1384" and apply your review comments.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,27 +1,76 @@
...
+        sh '''
+          mvn \
+            -f org.eclipse.xtext.maven.parent/pom.xml \
+            --batch-mode \
+            --update-snapshots \
+            -fae \
+            -PuseJenkinsSnapshots \
+            -DJENKINS_URL=$JENKINS_URL \
+            -Dmaven.test.failure.ignore=true \
+            -Dmaven.repo.local=.m2/repository \
```
Comment (cdietrich) [inline Jenkinsfile]: you still use a local path there. it must be absolute

Comment (cdietrich) [reply Jenkinsfile]: thus there was the `${workspace}` before

**Comment by kthoms** [comment]
Actually the absolute path is important, as you said. The pom.xml in org.eclipse.xtext.maven.plugin installs the plugin jar into `[WORKSPACE]/.m2/repository`. When executing the IT test the working directory is the module subdir, and the test execution tries to resolve the IT-SNASPHOT jar from `[WORKSPACE]/org.eclipse.xtext.maven.plugin/.m2/repository`, so _not_ the location it was installed to.

**Pull request merged**